localstack/services/opensearch/resource_providers/domain.py

```diff
@@ -0,0 +1,250 @@
+    Idp: Optional[Idp] = None
+    MasterBackendRole: Optional[str] = None
+    MasterUserName: Optional[str] = None
+    RolesKey: Optional[str] = None
+    SessionTimeoutMinutes: Optional[int] = None
+    SubjectKey: Optional[str] = None
+
+
+@dataclass
+class AdvancedSecurityOptions:
+    AnonymousAuthDisableDate: Optional[str] = None
+    AnonymousAuthEnabled: Optional[bool] = None
+    Enabled: Optional[bool] = None
+    InternalUserDatabaseEnabled: Optional[bool] = None
+    MasterUserOptions: Optional[MasterUserOptions] = None
+    SAMLOptions: Optional[SAMLOptions] = None
+
+
+@dataclass
+class EBSOptions:
+    EBSEnabled: Optional[bool] = None
+    Iops: Optional[int] = None
+    Throughput: Optional[int] = None
+    VolumeSize: Optional[int] = None
+    VolumeType: Optional[str] = None
+
+
+@dataclass
+class EncryptionAtRestOptions:
+    Enabled: Optional[bool] = None
+    KmsKeyId: Optional[str] = None
+
+
+@dataclass
+class ServiceSoftwareOptions:
+    AutomatedUpdateDate: Optional[str] = None
+    Cancellable: Optional[bool] = None
+    CurrentVersion: Optional[str] = None
+    Description: Optional[str] = None
+    NewVersion: Optional[str] = None
+    OptionalDeployment: Optional[bool] = None
+    UpdateAvailable: Optional[bool] = None
+    UpdateStatus: Optional[str] = None
+
+
+@dataclass
+class WindowStartTime:
+    Hours: Optional[int] = None
+    Minutes: Optional[int] = None
+
+
+@dataclass
+class OffPeakWindow:
+    WindowStartTime: Optional[WindowStartTime] = None
+
+
+@dataclass
+class OffPeakWindowOptions:
+    Enabled: Optional[bool] = None
+    OffPeakWindow: Optional[OffPeakWindow] = None
+
+
+@dataclass
+class SoftwareUpdateOptions:
+    AutoSoftwareUpdateEnabled: Optional[bool] = None
+
+
+@dataclass
+class OpenSearchDomainProperties:
+    AccessPolicies: Optional[dict] = None
+    AdvancedOptions: Optional[dict] = None
+    AdvancedSecurityOptions: Optional[AdvancedSecurityOptions] = None
+    Arn: Optional[str] = None
+    ClusterConfig: Optional[ClusterConfig] = None
+    CognitoOptions: Optional[CognitoOptions] = None
+    DomainArn: Optional[str] = None
+    DomainEndpoint: Optional[str] = None
+    DomainEndpointOptions: Optional[DomainEndpointOptions] = None
+    DomainEndpoints: Optional[dict] = None
+    DomainName: Optional[str] = None
+    EBSOptions: Optional[EBSOptions] = None
+    EncryptionAtRestOptions: Optional[EncryptionAtRestOptions] = None
+    EngineVersion: Optional[str] = None
+    Id: Optional[str] = None
+    LogPublishingOptions: Optional[dict] = None
+    NodeToNodeEncryptionOptions: Optional[NodeToNodeEncryptionOptions] = None
+    OffPeakWindowOptions: Optional[OffPeakWindowOptions] = None
+    ServiceSoftwareOptions: Optional[ServiceSoftwareOptions] = None
+    SnapshotOptions: Optional[SnapshotOptions] = None
+    SoftwareUpdateOptions: Optional[SoftwareUpdateOptions] = None
+    Tags: Optional[list] = None
+    VPCOptions: Optional[VPCOptions] = None
```

Comment:
We might have to extract those to a separate file per service to avoid a lot of redundant definitions for some services

Reply:
Yeah I think we can probably generate the resources for a service at once, and combine the types from all resources :+1: